### PR TITLE
Fix error reporter not reporting to database

### LINKED ISSUE #23035: Error reporter not reporting to database
**Original reporter:** Jos Cooper

This is high priority as it could hold up the release.

### Expected behavior

Information from the error reporter should be sent to the database

### Actual behavior

Information entered in the error reporter is not being collected on the database

### Steps to reproduce the behavior

* Open MantidPlot
* Crash MantidPlot with the Segfault algorithm
* Give a name and email address in the Error Reporter
* Look at the database to see if the information has been registered

### Platforms affected

All

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**
This fixes the issue where error reports were not reaching the database
**Report to:** keith. 

**To test:**
Check that the package version of Mantid successfully sends error reports.

Full details of the testing can be found in Pull Request #23058. One of the files added there is ErrorAndRecovery.rst, it contains the testing instructions.

Fixes #23035 . 

Does this update require release notes?
- [ ] Yes
- [x] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by keeeto — DISMISSED** [review]
I have tested according to a new unscripted set of tests of error reporting. This passes the technical tests and works as it should

File: Framework/Kernel/src/ErrorReporter.cpp [hunk 1]
```diff
@@ -99,10 +99,8 @@ std::string ErrorReporter::generateErrorMessage() {
 
   message["exitCode"] = m_exitCode;
 
-  if (m_share) {
-    message["email"] = m_email;
-    message["name"] = m_name;
-  }
+  message["email"] = m_email;
```
Comment (martyngigg) [inline Framework/Kernel/src/ErrorReporter.cpp]: Isn't this now sharing the name/email when it has not been asked too?

Comment (martyngigg) [reply Framework/Kernel/src/ErrorReporter.cpp]: I just noticed that clang says `m_share` is not used any more. Does this mean the share/no share stuff is handled differently now?

Comment (Mantid-Matthew) [reply Framework/Kernel/src/ErrorReporter.cpp]: @martyngigg 
This is due to the change made where the share only non identifying information is grayed out if anything is entered in the name or email boxes. This means that for this case the name and email we are sharing will always be empty strings. This does somewhat remove the need for m_share at all. The change was originally made becuase the web service was returning an error code if the post request it received didn't have a name and email field. 

I think there are two solutions which would be better than this time allowing
* Keep the if statement and explicitly set name and email to empty strings to reduce chance of future changes breaking confidentiality. 
* Revert this change entirely and update the web service to be able to deal with post requests which do not contain name and email.

Comment (martyngigg) [reply Framework/Kernel/src/ErrorReporter.cpp]: At this point I think I would opt for option 1. As you say, it's not guaranteed that the name/email is blank so I think forcing those fields blank when they are not requested indicates to future developers that there is a different situation to deal with here.

**Review by martyngigg — APPROVED** [review]
I'm happy with the code changes as they currently are.

**Review by keeeto — APPROVED** [review]
This passes all the unscripted tests on OSX

**Pull request merged**